identify/extensions.py

```diff
@@ -29,6 +29,7 @@
     'css': {'text', 'css'},
     'csv': {'text', 'csv'},
     'cu': {'text', 'cuda'},
+    'cuh': {'text', 'header', 'cuda'},
```

Comment:
just a few questions (also I would probably want a stronger source than a SO post -- perhaps something official from the CUDA project?)

- can all `.cuh` files be interpreted as C headers? otherwise the `'header'` tag might not be appropriate here
- can all `.cuh` files be interpreted the same as `.cu` files would be?

Reply:
@asottile Thanks for the review!

There are official CUDA sample codes and popular CUDA applications that use `.cuh`, here is a [specific example](https://github.com/NVIDIA/cuda-samples/tree/568b39bd5bea5924446d49d07a4c026a1ec945a5/Samples/shfl_scan) and [a longer list](https://github.com/NVIDIA/cuda-samples/search?q=cuh), as well as some NVIDIA provided libraries like [cub](https://github.com/NVIDIA/cub/tree/main/cub) and non-NVIDIA applications like [HOOMD-blue](https://github.com/glotzerlab/hoomd-blue/tree/master/hoomd).

- `.cuh` is used for CUDA C/C++ headers, so they're not plain C/C++. We can remove `'header'` if that's the intended distinction, I am just trying to follow the right pattern. 😄
- I'm not 100% sure what you're asking in the second question, but the general idea is that `.cuh` is to `.h` as `.cu` is to `.cc` or `.cpp`. A `.cuh` file is a header file included by a `.cu` file that may include CUDA C/C++ keywords and syntax.

I would be fine with this suggestion if it makes more sense to you.
```suggestion
    'cuh': {'text', 'cuda'},
```

Reply:
yeah that seems fine !

Reply:
Cool, I fixed that. 👍